lib/python-sdk/common_grants_sdk/schemas/base.py

```diff
@@ -18,13 +22,17 @@ def dump_json(self) -> str:
         """Convert model to JSON string (alias for model_dump_json for backward compatibility)."""
         return self.model_dump_json()
 
+    def dump_with_mapping(self, mapping: dict) -> dict:
+        """Convert model to dictionary with mapping."""
+        return transform_from_mapping(self.model_dump(mode="json"), mapping)
+
     @classmethod
-    def from_json(cls, json_str: str) -> "CommonGrantsBaseModel":
+    def from_json(cls, json_str: str) -> Self:
```

Comment:
This improves the intellisense output, so that the class method displays the name of the sub-class being instantiated (if it inherits from this base model) instead of always displaying "CommonGrantsBaseModel".